src/components/MessagesList/MessagesList.vue

```diff
@@ -329,40 +352,89 @@ export default {
 
 	methods: {
 		prepareMessagesGroups(messages) {
-			const groups = []
+			let prevGroupMap = {}
+			const groupsByDate = {}
 			let lastMessage = null
+			let groupId = null
+			let dateTimestamp = null
 			for (const message of messages) {
 				if (!this.messagesShouldBeGrouped(message, lastMessage)) {
-					if (groups.at(-1)) {
-						groups.at(-1).nextMessageId = message.id
+					groupId = message.id
+					if (message.timestamp === 0) {
+						// This is a temporary message, the timestamp is today
+						dateTimestamp = moment().startOf('day').unix()
+					} else {
+						dateTimestamp = moment(message.timestamp * 1000).startOf('day').unix()
+					}
+					if (!groupsByDate[dateTimestamp]) {
+						groupsByDate[dateTimestamp] = {}
 					}
 
-					groups.push({
+					groupsByDate[dateTimestamp][groupId] = {
 						id: message.id,
 						messages: [message],
-						dateSeparator: this.generateDateSeparator(message, lastMessage),
-						previousMessageId: groups.at(-1)?.messages.at(-1).id ?? 0,
+						token: this.token,
+						dateTimestamp,
```

Comment:
```suggestion
```

is token needed? I couldn't find where it is used
for dateTimestamp see another comments. Seems like also not needed anymore in softUpdate

Reply:
I know but I prefer to keep it, maybe it is needed in the future to conserver its uniqueness ( id+token) .